### DISCUSSION TITLE
How to run a infinite loop task in backgroud?

### DISCUSSION THREAD

**Discussion by 0106WeiWeiDeng** [aio-libs/aiohttp#7236] [Q&A]
Hi,
Recently, I design a analysis platform using aiohttp and slurm, and I want to use a infinite loop to check slurm job state in backgroud, the code looks like this:

in view.py
```
async def query_slurm_job_state(app):
    while True:
        await asyncio.sleep(600) # run one time per 10 min 
        async with lock: # lock for slurm_jobs dict
           async with app['mysql_db'].acquire() as conn:
              for key in slurm_jobs:
                 # query slurm job state, and updatae it in mysql
```

in main.py

```
app = web.Application()
# how to run query_slurm_job_state function in backgroud?
web.run_app(app)
```

How to run query_slurm_job_state function in backgroud? Can I use `app.on_startup.append()` ? Does anyone give me a hint? Thanks in advance!

**Comment by Dreamsorcerer** [comment]
https://docs.aiohttp.org/en/stable/web_advanced.html#background-tasks
